baseapp-chats/baseapp_chats/graphql/interfaces.py

```diff
@@ -0,0 +1,46 @@
+import graphene
+import swapper
+from baseapp_core.graphql import get_object_type_for_model
+from django.db.models import Sum
+from graphene import relay
+from graphene_django.filter import DjangoFilterConnectionField
+
+ChatRoom = swapper.load_model("baseapp_chats", "ChatRoom")
+UnreadMessageCount = swapper.load_model("baseapp_chats", "UnreadMessageCount")
+Block = swapper.load_model("baseapp_blocks", "Block")
+
+
+class ChatRoomsInterface(relay.Node):
+    chat_rooms = DjangoFilterConnectionField(get_object_type_for_model(ChatRoom))
+    unread_messages_count = graphene.Int()
+
+    def resolve_chat_rooms(self, info, **kwargs):
+        if not info.context.user.has_perm("baseapp_chats.list_chatrooms", self):
+            return ChatRoom.objects.none()
+
+        qs = ChatRoom.objects.filter(
+            participants__profile_id=self.pk,
+        ).order_by("-last_message_time", "-created")
+
+        # Exclude rooms where the profile is blocked
+        blocking_profile_ids = Block.objects.filter(actor_id=self.pk).values_list(
```

Comment:
is self.pk a profile pk ?

Reply:
yeah it should be. Actually this could be different instances depending on the [object type](https://docs.graphene-python.org/en/latest/types/objecttypes/)  in which `ChatRoomsInterface` has been added to the class meta [interfaces](https://docs.graphene-python.org/en/latest/types/interfaces/)

Reply:
on this implementation only the Profile can implement `ChatRoomsInterface`